packages/playground/src/utils/ssh.ts

```diff
@@ -0,0 +1,232 @@
+import { InsufficientBalanceError } from "@threefold/types";
+import crypto from "crypto";
+
+import { useProfileManager } from "@/stores/profile_manager";
+import type { SSHKeyData } from "@/types";
+
+import { createCustomToast, ToastType } from "./custom_toast";
+import { getGrid, loadBalance, storeSSH } from "./grid";
+import { downloadAsJson } from "./helpers";
+
+/**
+ * Manages SSH key operations including migration, updating, exporting, deleting, and listing.
+ */
+class SSHKeysManagement {
+  private oldKey = "";
+  updateCost = 0.01;
+  private words = [
+    "moon",
+    "earth",
+    "sun",
+    "star",
+    "galaxy",
+    "nebula",
+    "comet",
+    "planet",
+    "asteroid",
+    "satellite",
+    "mercury",
+    "venus",
+    "mars",
+    "jupiter",
+    "saturn",
+    "uranus",
+    "neptune",
+    "pluto",
+    "meteor",
+    "cosmos",
+  ];
+
+  constructor() {
+    const profileManager = useProfileManager();
+    this.oldKey = profileManager.profile?.ssh as unknown as string;
+  }
+
+  /**
+   * Migrates an old SSH key string to the new SSHKeyData format.
+   * @returns An array containing the migrated SSHKeyData.
+   */
+  migrate(): SSHKeyData[] {
+    const userKeys: SSHKeyData[] = [];
+
+    let keyName = "";
+    const parts = this.oldKey.split(" ");
+
+    if (parts.length < 3) {
+      keyName = this.generateName()!;
+    } else {
+      keyName = parts[parts.length - 1];
+    }
+    const newKey: SSHKeyData = {
+      createdAt: this.formatDate(new Date()),
+      name: keyName,
+      id: 1,
+      isActive: true,
+      publicKey: this.oldKey,
+    };
+    userKeys.push(newKey);
+    return userKeys;
+  }
+
+  /**
+   * Checks if the SSH key has not been migrated yet.
+   * @returns A boolean indicating whether the key has not been migrated.
+   */
+  migrated(): boolean {
+    return typeof this.oldKey !== "string";
+  }
+
+  /**
+   * Updates SSH keys in the profile and stores them in the grid.
+   * @param keys The SSH keys to update.
+   */
+  async update(keys: SSHKeyData[]): Promise<void> {
+    const profileManager = useProfileManager();
+    const grid = await getGrid(profileManager.profile!);
+    if (!grid) {
+      createCustomToast(`Error occurred because the grid has not initialized yet.`, ToastType.danger);
+      return;
+    }
+
+    const balance = await loadBalance(grid!);
+    if (balance.free === balance.locked || balance.free < this.updateCost) {
+      throw new Error(
+        "Your wallet balance is insufficient to save your SSH key. To avoid losing your SSH key, please recharge your wallet.",
+      );
+    }
+
```

Comment:
I think there is no need to check for balance and throw errror here, the client already do that for us, 
and this is the idea of using types package and errror classes, client throw an error  then we customize the displayed error massage for the user based on the thrown error class.

Reply:
Yeah, you have a point, errors should be handled outside the management class, However, it is only used in the SSH component and not planned to be used somewhere else, I think we can keep all SSH-related in one place.